internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -185,6 +222,113 @@ func (a *runTaskAction) Do(o *Orchestrator) error {
 	return nil
 }
 
+// setupProxyConnections creates proxy connections to a.hosts in pauseContainer.
+// It assumes that pauseContainer is already running. A unique proxy connection
+// is created for each host (in parallel) using AWS SSM Port Forwarding through
+// a.ssmTarget. Then, each connection is assigned an IP from a.network,
+// starting at the bottom of the IP range. Using iptables, TCP packets destined
+// for the connection's assigned IP are redirected to the connection. Finally,
+// the host's name is mapped to its assigned IP in /etc/hosts.
+func (o *Orchestrator) setupProxyConnections(ctx context.Context, pauseContainer string, a *runTaskAction) error {
+	fmt.Printf("\nSetting up proxy connections...\n")
+
+	ports := make(map[Host]uint16)
+	port := uint16(50000)
+	for i := range a.hosts {
+		ports[a.hosts[i]] = port
+		port++
+	}
+
+	for _, host := range a.hosts {
+		host := host
+		portForHost := ports[host]
+
+		o.wg.Add(1)
+		go func() {
+			defer o.wg.Done()
+
+			err := o.docker.Exec(context.Background(), pauseContainer, io.Discard, "aws", "ssm", "start-session",
```

Comment:
Do we need to describe anything or the session will always start once this cmd execution succeeds?

Reply:
i _think_ it will always start - the command will exit if it fails to establish the connection, at which point we'll shut down.